### PR TITLE
chore(ci): add code coverage support

### PR DESCRIPTION
**Pull request by evansims**
## Description

This PR introduces Codecov to our CI for monitoring test coverage of changes.

## Review Checklist
- [x] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [x] The correct base branch is being used, if not `main`
- [ ] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

**Comment by evansims** [comment]
Looks like govuln spotted an unrelated dependency issue. I'll cut a separate PR for that.

File: .github/workflows/main.yaml [hunk 1]
```diff
@@ -10,39 +10,46 @@ jobs:
...
+          slug: openfga/go-sdk
 
   create-release:
     runs-on: ubuntu-latest
     if: startsWith(github.ref, 'refs/tags/v')
     needs: [test]
 
     steps:
-      - uses: actions/checkout@9bb56186c3b09b4f86b1c65136769dd318469633 # v4.5.4
+      - uses: actions/checkout@9bb56186c3b09b4f86b1c65136769dd318469633 # v4.1.2
```
Comment (rhamzeh) [inline .github/workflows/main.yaml]: Looking at https://github.com/actions/checkout/releases, this seems to be a pre-release

Comment (rhamzeh) [reply .github/workflows/main.yaml]: Considering we don't use sparse checkout, I'm OK with it

**Review by rhamzeh — DISMISSED** [review]

**Comment by rhamzeh** [comment]
To get the sec issue to resolve, please increment [this](https://github.com/openfga/go-sdk/blob/main/.github/workflows/main.yaml#L21) to `>= 1.22.2`

File: .github/workflows/main.yaml (hunk 1 above)
Comment (evansims) [reply .github/workflows/main.yaml]: Thanks for catching that!

**Review by rhamzeh — APPROVED** [review]

**Pull request merged**